### PR TITLE
Add a sub array function to Array and test

### PR DESCRIPTION
**Pull request by Dendim0n**
add two new functions:

    ///EZSE: Get a sub array from range of index
    public func subArray(fromIndex:Int,toIndex:Int) -> Array
    
    ///EZSE: Get a sub array from elements that meet specific condition
    public func subArray(_ condition: (Element) -> Bool) -> Array

### CODE REVIEW

File: Sources/ArrayExtensions.swift [hunk 1]
```diff
@@ -19,6 +19,26 @@ public func ==<T: Equatable>(lhs: [T]?, rhs: [T]?) -> Bool {
 }
 
 extension Array {
+    
+    ///EZSE: Get a sub array from range of index
+    public func subArray(fromIndex:Int,toIndex:Int) -> Array {
```
Comment (piv199) [inline Sources/ArrayExtensions.swift]: Why this is better than `Array(array[fromIndex...toIndex])`

File: Sources/ArrayExtensions.swift [hunk 2]
```diff
@@ -19,6 +19,26 @@ public func ==<T: Equatable>(lhs: [T]?, rhs: [T]?) -> Bool {
...
+        var subArray = Array()
+        for index in fromIndex...toIndex {
+            subArray.append(self[index])
+        }
+        return subArray
+    }
+    
+    ///EZSE: Get a sub array from elements that meet specific condition
+    public func subArray(_ condition: (Element) -> Bool) -> Array {
```
Comment (piv199) [inline Sources/ArrayExtensions.swift]: This is copy of `array.filter` function.

File: Sources/ArrayExtensions.swift [hunk 3]
```diff
@@ -19,6 +19,15 @@ public func ==<T: Equatable>(lhs: [T]?, rhs: [T]?) -> Bool {
 }
 
 extension Array {
+    
+    ///EZSE: Get a sub array from range of index
+    public func get(atIndex:Int,toIndex:Int) -> Array {
```
Comment (piv199) [inline Sources/ArrayExtensions.swift]: think of `get(at range: ClosedRange<Int>) -> Array` 
And also you are missing checks for lower bound and upper bound
Thirdly you could implement it like `return self[range]`

**Comment by piv199** [comment]
Also each new extension should have tests for it, we are currently trying to increase our code coverage.

Thanks for your pull request.
We are always appriciate any help.

**Comment by lfarah** [comment]
@Dendim0n let us know if you need help writing the tests! 😉

**Comment by piv199** [comment]
Could you please reset `Dendim0n:master` to `0ec8f56` commit?

Some refs:
[StackOverflow](http://stackoverflow.com/questions/17667023/git-how-to-reset-origin-master-to-a-commit)

**Comment by piv199** [comment]
@lfarah could we approve it?

**Comment by lfarah** [comment]
@piv199 I didn't like that ```testGet_SubArray``` with the underscore...

**Comment by lfarah** [comment]
@piv199 sent you a message on Gitter about it

**Comment by lfarah** [comment]
@Dendim0n thank you so much for helping out! 😃 👍

**Pull request merged**